solr/core/src/java/org/apache/solr/search/facet/FacetRangeProcessor.java

```diff
@@ -774,6 +763,30 @@ public Long parseAndAddGap(@SuppressWarnings("rawtypes") Comparable value, Strin
     }
   }
 
+  private static class EnumCalc extends Calc {
+
+    private final EnumMapping mapping;
+    public EnumCalc(final SchemaField f) {
+      super(f);
+      mapping = ((AbstractEnumField)field.getType()).getEnumMapping();
+    }
+    @Override
+    public EnumFieldValue bitsToValue(long bits) {
+      Integer val = (int)bits;
+      return new EnumFieldValue(val, mapping.intValueToStringValue(val));
+    }
+    @Override
+    protected EnumFieldValue parseStr(String rawval) {
```

Comment:
The test didn't exercise this.  The impl looks suspicious because it parses a String to an integer.

Reply:
It is a bit difficult to test it directly, the respone contains EnumFieldValue instances which are convert to to the string value by writing out the response as XML by TestHarness query response writer. Before the change the int values are written out as string, so the test at least test a part of the expected behavior

Reply:
Added test to verify correct instance of results

Reply:
I already added an additional request to work directly on the objects and check the instanceof EnumFieldValue